golos.vesting/golos.vesting.cpp

```diff
@@ -74,6 +74,44 @@ void vesting::on_transfer(name from, name to, asset quantity, std::string memo)
     add_balance(recipient != name() ? recipient : from, converted, has_auth(to) ? to : from);
 }
 
+void vesting::on_transfer_vesting(name from, name to, asset quantity, std::string memo) {
+    if (_self != to)
+        return;
+
+    auto recipient = get_recipient(memo);
+    if (token::get_issuer(config::token_name, quantity.symbol.code()) == from && recipient == name())
+        return;     // just increase token supply
+
+    add_balance(recipient != name() ? recipient : from, quantity, has_auth(to) ? to : from);
+}
+
+void vesting::on_bulk_transfer(name from, std::vector<token::recipient> recipients) {
+    auto token_symbol = recipients.at(0).quantity.symbol;
+
+    asset sum_quantity(0, token_symbol);
+    for (auto recipient_obj : recipients) {
+        sum_quantity += recipient_obj.quantity;
+    }
+
+    vesting_table table_vesting(_self, _self.value);    // TODO: use symbol as scope #550
+    auto vesting = table_vesting.find(token_symbol.code().raw());
+    eosio_assert(vesting != table_vesting.end(), "Token not found");
+    asset converted = token_to_vesting(sum_quantity, *vesting, -sum_quantity.amount);
+
+    uint64_t sum_quantity_recipient = 0;
+    for (auto recipient_obj : recipients) {
+        auto quantity_recipient = static_cast<uint64_t>(converted.amount * (static_cast<double>(recipient_obj.quantity.amount) / sum_quantity.amount));
+        sum_quantity_recipient += quantity_recipient;
+        on_transfer_vesting(from, recipient_obj.to, asset(quantity_recipient, converted.symbol), recipient_obj.memo);
+    }
+
+    table_vesting.modify(vesting, name(), [&](auto& item) {
+        item.supply += asset(sum_quantity_recipient, converted.symbol);
```

Comment:
this value can still be wrong if `on_transfer_vesting` exits early at least once (`_self != to`). so recipients, who are not `_self` should also be skipped in calculation of `sum_quantity`.
please add test to ensure this case handled properly